### ISSUE TITLE
iOS build error: Import of module 'glog.glog.log_severity' appears within namespace 'google'

### ISSUE THREAD

**Issue by iArchin** [facebook/react-native#41045]
### Description

I already know that there are other similar issues of this with multiple way of fix but I'm pretty sure that I have tried them but not worked. this is how my pod file looks: 

```
# Resolve react_native_pods.rb with node to allow for hoisting
require Pod::Executable.execute_command('node', ['-p',
  'require.resolve(
    "react-native/scripts/react_native_pods.rb",
    {paths: [process.argv[1]]},
  )', __dir__]).strip

platform :ios, min_ios_version_supported
prepare_react_native_project!

pod 'GoogleUtilities', :modular_headers => true

# If you are using a `react-native-flipper` your iOS build will fail when `NO_FLIPPER=1` is set.
# because `react-native-flipper` depends on (FlipperKit,...) that will be excluded
#
# To fix this you can also exclude `react-native-flipper` using a `react-native.config.js`
# ```js
# module.exports = {
#   dependencies: {
#     ...(process.env.NO_FLIPPER ? { 'react-native-flipper': { platforms: { ios: null } } } : {}),
# ```
flipper_config = ENV['NO_FLIPPER'] == "1" ? FlipperConfiguration.disabled : FlipperConfiguration.enabled
$RNMapboxMapsImpl = 'mapbox'

linkage = ENV['USE_FRAMEWORKS']
if linkage != nil
  Pod::UI.puts "Configuring Pod with #{linkage}ally linked Frameworks".green
  use_frameworks! :linkage => linkage.to_sym
end

target 'myAwesomeApp' do
  config = use_native_modules!

  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    # Hermes is now enabled by default. Disable by setting this flag to false.
    # Upcoming versions of React Native may rely on get_default_flags(), but
    # we make it explicit here to aid in the React Native upgrade process.
    :hermes_enabled => flags[:hermes_enabled],
    :fabric_enabled => flags[:fabric_enabled],
    # Enables Flipper.
    #
    # Note that if you have use_frameworks! enabled, Flipper will not work and
    # you should disable the next line.
    :flipper_configuration => flipper_config,
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  target 'myAwesomeAppTests' do
    inherit! :complete
    # Pods for testing
  end

  pre_install do |installer|
    $RNMapboxMaps.pre_install(installer)
  end

  post_install do |installer|
    $RNMapboxMaps.post_install(installer)
    react_native_post_install(
      installer,
      # Set `mac_catalyst_enabled` to `true` in order to apply patches
      # necessary for Mac Catalyst builds
      :mac_catalyst_enabled => false
    )
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
  end
end
```
my package.json file: 
```

{
  "name": "myAwesomeApp",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "lint": "eslint . --ext .js,.jsx,.ts,.tsx",
    "lint:fix": "eslint . --ext .js,.jsx,.ts,.tsx --fix",
    "start": "react-native start",
    "test": "jest",
    "postinstall": "patch-package"
  },
  "dependencies": {
    "@gorhom/bottom-sheet": "^4",
    "@microsoft/signalr": "^7.0.7",
    "@react-native-async-storage/async-storage": "^1.18.2",
    "@react-native-community/slider": "^4.4.2",
    "@react-native-firebase/app": "^18.0.0",
    "@react-native-firebase/auth": "^18.3.0",
    "@react-native-firebase/messaging": "^18.0.0",
    "@react-native/metro-config": "^0.72.11",
    "@react-navigation/bottom-tabs": "^6.5.7",
    "@react-navigation/native": "^6.1.6",
    "@react-navigation/native-stack": "^6.9.12",
    "@react-navigation/stack": "^6.3.16",
    "@reduxjs/toolkit": "^1.9.5",
    "@rnmapbox/maps": "^10.0.12",
    "@sentry/react-native": "^5.11.0",
    "@shopify/flash-list": "^1.5.0",
    "axios": "^1.4.0",
    "dotenv": "^16.3.1",
    "lodash.debounce": "^4.0.8",
    "moment": "^2.29.4",
    "qs": "^6.11.2",
    "react": "18.2.0",
    "react-native": "0.72.2",
    "react-native-calendar-picker": "^7.1.4",
    "react-native-config": "^1.5.1",
    "react-native-confirmation-code-field": "^7.3.1",
    "react-native-fast-image": "^8.6.3",
    "react-native-feather": "^1.1.2",
    "react-native-geolocation-service": "^5.3.1",
    "react-native-gesture-handler": "^2.10.2",
    "react-native-gradle-plugin": "^0.71.19",
    "react-native-image-viewing": "^0.2.2",
    "react-native-linear-gradient": "^2.7.3",
    "react-native-modal": "^13.0.1",
    "react-native-reanimated": "^3.3.0",
    "react-native-reanimated-carousel": "^3.4.0",
    "react-native-safe-area-context": "^4.5.3",
    "react-native-screens": "^3.20.0",
    "react-native-shared-element": "^0.8.8",
    "react-native-snap-carousel": "^3.9.1",
    "react-native-svg": "^13.9.0",
    "react-native-toast-message": "^2.1.6",
    "react-navigation-shared-element": "^3.1.3",
    "react-query": "^3.39.3",
    "react-redux": "^8.1.0",
    "redux": "^4.2.1",
    "redux-persist": "^6.0.0",
    "remove": "^0.1.5"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@babel/preset-env": "^7.20.0",
    "@babel/runtime": "^7.20.0",
    "@react-native-community/eslint-config": "^3.2.0",
    "@tsconfig/react-native": "^3.0.2",
    "@types/jest": "^29.5.2",
    "@types/lodash.debounce": "^4.0.7",
    "@types/react": "^18.2.8",
    "@types/react-native-calendar-picker": "^7.0.3",
    "@types/react-test-renderer": "^18.0.0",
    "babel-jest": "^29.2.1",
    "babel-plugin-module-resolver": "^5.0.0",
    "eslint": "^8.19.0",
    "eslint-plugin-import": "^2.27.5",
    "jest": "^29.2.1",
    "metro-react-native-babel-preset": "0.73.9",
    "patch-package": "^7.0.2",
    "postinstall-postinstall": "^2.1.0",
    "prettier": "^2.4.1",
    "react-native-flipper": "^0.211.0",
    "react-test-renderer": "18.2.0",
    "redux-flipper": "^2.0.2",
    "typescript": "^5.1.3"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

### React Native Version

0.72.2

### Output of `npx react-native info`

System:
  OS: macOS 12.7
  CPU: (4) x64 Intel(R) Core(TM) i5-5350U CPU @ 1.80GHz
  Memory: 27.18 MB / 8.00 GB
  Shell:
    version: 5.8.1
    path: /bin/zsh
Binaries:
  Node:
    version: 18.18.1
    path: /usr/local/bin/node
  Yarn:
    version: 1.22.19
    path: /usr/local/bin/yarn
  npm:
    version: 9.8.1
    path: /usr/local/bin/npm
  Watchman:
    version: 2023.10.09.00
    path: /usr/local/bin/watchman
Managers:
  CocoaPods:
    version: 1.13.0
    path: /usr/local/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.2
      - iOS 16.2
      - macOS 13.1
      - tvOS 16.1
      - watchOS 9.1
  Android SDK: Not Found
IDEs:
  Android Studio: Not Found
  Xcode:
    version: 14.2/14C18
    path: /usr/bin/xcodebuild
Languages:
  Java: Not Found
  Ruby:
    version: 2.7.0
    path: /Users/mac/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false

### Steps to reproduce

I provide my RN version and my packages also cocoapods version and ruby. 

### Snack, screenshot, or link to a repository

![Screenshot from 2023-10-18 11-06-16](https://github.com/facebook/react-native/assets/45012003/872742cd-9856-4f56-ba46-68d6e37c7d97)

**Comment by cortinico** [comment]
> I already know that there are other similar issues of this with multiple way of fix but I'm pretty sure that I have tried them but not worked. this is how my pod file looks:

Which other issues are you referring to?

**Comment by iArchin** [comment]
> > I already know that there are other similar issues of this with multiple way of fix but I'm pretty sure that I have tried them but not worked. this is how my pod file looks:
> 
> Which other issues are you referring to?

I meant [This](https://github.com/google/glog/issues/393) and #26217. 
but actually I found the solution in #38294. I upgrade my RN version from `0.72.2 `to `0.72.3` with `XCode 14.2` it solved.

**Comment by cortinico** [comment]
Closing as solved

**Comment by 592da** [comment]
Also happened to me with 0.73.1
XCode 14.3
@cortinico any thoughts?

**Comment by CapleaIonut-Alexandru** [comment]
I have the same issue on my expo project. Somebody know how to fix it ?

**Comment by paulschreiber** [comment]
@592da Did you figure this out for 0.73.x?

**Comment by nagasailokesh-toddle** [comment]
same issue. Somebody know how to fix it?

**Comment by Mahdisaghroun** [comment]
<img width="1149" alt="Screenshot 2024-02-12 at 09 38 43" src="https://github.com/facebook/react-native/assets/50046296/e331040a-98d3-498d-9b11-6925b5de90ce">

react-native: 0.73.4

**Comment by paulschreiber** [comment]
@kelset Any suggestions? This is still a problem in 0.73.4 for several folks.

**Comment by kelset** [comment]
@paulschreiber I'm currently not on the release crew but I would recommend opening a new issue that provides a repro so that the team can look into this issue and why it keeps resurfacing

**Comment by paulschreiber** [comment]
@kelset I don't have time to write up a new mini app with a repo case. I did get this working eventually. Some combination of (disabling flipper/use_frameworks! :linkage => :static) and (upgrading to macOS 14/Xcode 15).

**Comment by Victor-Nyagudi** [comment]
If you're still experiencing this issue, consider trying this [StackOverflow answer](https://stackoverflow.com/a/77994015/15890330).

**Comment by codding123vbf** [comment]
> Closing as solved

im still getting this error

**Comment by codding123vbf** [comment]
> If you're still experiencing this issue, consider trying this [StackOverflow answer](https://stackoverflow.com/a/77994015/15890330).

it didnt fix my error man...

**Comment by codding123vbf** [comment]
after installing pods and running my project on xcode i started getting this error and its not solved yet my xcode version is 14.0.1...pls help

**Comment by codding123vbf** [comment]
i have fixed this issue ....if anyone is still facing this issue reach me i will help u with it

**Comment by paulschreiber** [comment]
@codding123vbf what was your fix? My fix was upgrading my GitHub runner form macos-12 to macos-14.

**Comment by barrownicholas** [comment]
If you're using a github action, it seems that `macos-latest` is still on v12 (despite the Docs saying otherwise); just use:
```yml
jobs:
  build:
    name: build
    runs-on: macos-14
```

**Comment by barrownicholas** [comment]
For visibility, I documented this a bit better here on SO: https://stackoverflow.com/a/78359974/14717625

**Comment by bhandanyan-nomad** [comment]
I'm getting this error for the first time on react-native v0.76.1

**Comment by rohitpareek147-rp** [comment]
i am also getting same issue on react-native 0.76.1 version and tried all solution available on net. 

@paulschreiber @cortinico  do you guys have any solution.

**Comment by rohitpareek147-rp** [comment]
`# Resolve react_native_pods.rb with node to allow for hoisting
#### react native  .072 + update for react-native-permissions
def node_require(script)
   # Resolve script with node to allow for hoisting
  require Pod::Executable.execute_command('node', ['-p',
     "require.resolve(
      '#{script}',
       {paths: [process.argv[1]]},
    )", __dir__]).strip
 end

node_require('react-native/scripts/react_native_pods.rb')
node_require('react-native-permissions/scripts/setup.rb')

platform :ios, min_ios_version_supported
prepare_react_native_project!
$toggleADEUMRNStaticFramework = true

setup_permissions([
  'Calendars',
  'Camera',
  'Contacts',
  'LocationAccuracy',
  'LocationAlways',
  'LocationWhenInUse',
  'MediaLibrary',
  'Microphone',
  'Notifications',
  'PhotoLibrary',
  'PhotoLibraryAddOnly',
])

linkage = ENV['USE_FRAMEWORKS']
if linkage != nil
  Pod::UI.puts "Configuring Pod with #{linkage}ally linked Frameworks".green
  use_frameworks! :linkage => linkage.to_sym
end

target 'kmbmobile' do
  config = use_native_modules!
  # use_frameworks! :linkage => :static
  $RNFirebaseAsStaticFramework = true  

  pre_install do |installer|
    installer.pod_targets.each do |pod|
      if pod.name.eql?('vision-camera-code-scanner') || pod.name.eql?('VisionCamera')
        def pod.build_type
          Pod::BuildType.static_library
        end
      end
    end
  end

  use_react_native!(
    :path => config[:reactNativePath],
    :hermes_enabled => true,
    :fabric_enabled => true,
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )
  
  pod 'react-native-config/Extension', :path => '../node_modules/react-native-config'
  pod 'clevertap-react-native', :path => '../node_modules/clevertap-react-native'

 pod 'FirebaseCoreInternal', :modular_headers => true 
 pod 'FirebaseCrashlytics', :modular_headers => true 
 pod 'FirebaseSessions', :modular_headers => true
 # Add modular headers for specific dependencies
 pod 'GoogleUtilities', :modular_headers => true 
 pod 'GoogleDataTransport', :modular_headers => true 
 pod 'nanopb', :modular_headers => true
 pod 'FirebaseCore', :modular_headers => true
 pod 'FirebaseInstallations', :modular_headers => true
 pod 'FirebaseCoreExtension', :modular_headers => true
#  pod 'RCT-Folly', :podspec => '../node_modules/react-native/third-party-podspecs/RCT-Folly.podspec'
 pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
 # pod 'RCTTypeSafety', :path => "../node_modules/react-native/Libraries/TypeSafety", :modular_headers => false
 # pod 'FBLazyVector', :path => "../node_modules/react-native/Libraries/FBLazyVector", :modular_headers => false


 
 
 post_install do |installer|
   #------------------ react/bridging/CallbackWrapper.h' file not found
   installer.pods_project.targets.each do |target|
     target.build_configurations.each do |config|
       config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '15.1'
       config.build_settings['HEADER_SEARCH_PATHS'] ||= '$(inherited) '
       config.build_settings['HEADER_SEARCH_PATHS'] << '"${PODS_ROOT}/../../node_modules/react-native/ReactCommon" '
       # if target.name === "RNReanimated"
       #   config.build_settings['CLANG_CXX_LANGUAGE_STANDARD'] = 'c++17'
       # end
     end
   end
   #------------------ pod changes added for issue-> react/bridging/CallbackWrapper.h' file not found Vision Camera
   # https://github.com/facebook/react-native/blob/main/packages/react-native/scripts/react_native_pods.rb#L197-L202
   react_native_post_install(
     installer,
     config[:reactNativePath],
     :mac_catalyst_enabled => false,
     # :ccache_enabled => true
   )
 end
end`

**Comment by rohitpareek147-rp** [comment]
> The following change (along with some import tweaks in AppDelegate.h) fixed this build issue in one of my projects
> 
> ```diff
> -  8EC73EE827FDCEB400945966 /* AppDelegate.mm in Sources */ = {isa = PBXBuildFile; fileRef = 8EC73EE727FDCEB400945966 /* AppDelegate.mm */; settings = {COMPILER_FLAGS = "-fmodules -fcxx-modules"; }; };
> +  8EC73EE827FDCEB400945966 /* AppDelegate.mm in Sources */ = {isa = PBXBuildFile; fileRef = 8EC73EE727FDCEB400945966 /* AppDelegate.mm */; };
> ```

@benhandanyan  Thanks for reply. could please elaborate solution in detail ? what tweaks have you done in AppDelegate.h.

**Comment by benhandanyan** [comment]
I misspoke, the import updates were in `AppDelegate.mm`. Update any `@import PackageName` to `#import PackageName`

**Comment by rohitpareek147-rp** [comment]
> The following change (along with some import tweaks in ~AppDelegate.h~ AppDelegate.mm) fixed this build issue in one of my projects
> 
> ```diff
> -  8EC73EE827FDCEB400945966 /* AppDelegate.mm in Sources */ = {isa = PBXBuildFile; fileRef = 8EC73EE727FDCEB400945966 /* AppDelegate.mm */; settings = {COMPILER_FLAGS = "-fmodules -fcxx-modules"; }; };
> +  8EC73EE827FDCEB400945966 /* AppDelegate.mm in Sources */ = {isa = PBXBuildFile; fileRef = 8EC73EE727FDCEB400945966 /* AppDelegate.mm */; };
> ```

Thanks @benhandanyan , the solution you suggested is worked for me 👍

**Comment by anujsachan1990** [comment]
I am having the same issue @rohitpareek147-rp  . I can't remove -fcxx-modules as this is a requirement for one of the packages I am using it keeps giving me 
iOS build error: Import of module 'glog.glog.log_severity' appears within namespace 'google' 

Xcode :- 16.2
React-native :- 0.76.1
React :- 18.3.1

did you manage to solve it a differtn way?

**Comment by nashht** [comment]
> I am having the same issue [@rohitpareek147-rp](https://github.com/rohitpareek147-rp) . I can't remove -fcxx-modules as this is a requirement for one of the packages I am using it keeps giving me iOS build error: Import of module 'glog.glog.log_severity' appears within namespace 'google'
> 
> Xcode :- 16.2 React-native :- 0.76.1 React :- 18.3.1
> 
> did you manage to solve it a differtn way?

same to you, please tell me the solution if you find it

**Comment by anujsachan1990** [comment]
@nashht 
To resolve this issue, I checked which module required -fcxx-modules. In my case, it was [Adobe's AEPSDK React Native](https://github.com/adobe/aepsdk-react-native).

I also reviewed their implementation to see if any updates had been made. They recently removed @import statements from the AppDelegate.h file and moved them into separate header files. You can see an example of this change in [this comment](https://github.com/adobe/aepsdk-react-native/issues/247#issuecomment-1462326054).

If you're facing a similar issue, I recommend checking whether the module you're using requires -fcxx-modules and reviewing any recent changes to its implementation.

**Comment by ankitpunchh** [comment]
This same issue we are getting on latest RN upgrade which 0.79.2. Anyone was able to fix it yet ?

**Comment by No-code-no-futrue** [comment]
I’m running into the same issue—it’s appeared across several versions already. Because I have to build with the -fcxx-modules flag for third-party compatibility, I can’t avoid it. Is there any fix or workaround?

**Comment by vinova-vito** [comment]
I’m running into the same issue—it’s appeared across several versions already. Because I have to build with the -fcxx-modules flag for third-party compatibility, I can’t avoid it. Is there any fix or workaround?

**Comment by vinova-vito** [comment]
> > I’m running into the same issue—it’s appeared across several versions already. Because I have to build with the -fcxx-modules flag for third-party compatibility, I can’t avoid it. Is there any fix or workaround?
> 
> Would you happen to know which third party needs -fcxx-modules flag? If yes check their latest on the React Native version you are using, if they have a different way of implementation

the third-party library is a vendor-custom iOS framework that uses @import. The project won’t build unless I pass the -fcxx-modules flag. I’m currently on React Native 0.79.1.

**Comment by ankitpunchh** [comment]
Hello everyone,

I encountered this issue while upgrading to React Native 0.79.2. I resolved it by removing -fcxx-modules from the build settings. However, this caused errors in libraries that depend on -fcxx-modules.

I updated most of the affected libraries, but one still relies on Objective-C modules. To fix that, I had to replace @import ModuleName with #import <ModuleName/ModuleName.h>. This workaround may vary depending on the specific library you're using.

**Comment by tw-anuj** [comment]
I am also facing the same issue. React native version - 0.79.0

<img width="452" alt="Image" src="https://github.com/user-attachments/assets/f6e39389-b66e-4fce-8389-23c0a0d390da" />

@ankitpunchh @anujsachan1990  Did you guys found any resolution ?

In my case also, I cannot remove -fcxx-modules as adobe packages are dependent on it.

This issue should be reopened.

**Comment by anujsachan1990** [comment]
> I am also facing the same issue. React native version - 0.79.0
> 
> <img alt="Image" width="452" src="https://private-user-images.githubusercontent.com/194590623/446720055-f6e39389-b66e-4fce-8389-23c0a0d390da.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDc5NTk3ODMsIm5iZiI6MTc0Nzk1OTQ4MywicGF0aCI6Ii8xOTQ1OTA2MjMvNDQ2NzIwMDU1LWY2ZTM5Mzg5LWI2NmUtNGZjZS04Mzg5LTIzYzBhMGQzOTBkYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUyM1QwMDE4MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYjRmYWY0NDA0ZjhjZGNhODQzMGViNDExMWZiM2ZiODQ5ODMzN2U1OTk1N2FjZWZiYjNhOWI3MDVlNmRiNTRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.MHIZNnGXTMNDGXaRlMz5PuzSWv8k9MY0iAS-lSynlCs">
> [@ankitpunchh](https://github.com/ankitpunchh) [@anujsachan1990](https://github.com/anujsachan1990) Did you guys found any resolution ?
> 
> In my case also, I cannot remove -fcxx-modules as adobe packages are dependent on it.
> 
> This issue should be reopened.

https://github.com/facebook/react-native/issues/41045#issuecomment-2730731092

**Comment by ankitpunchh** [comment]
> I am also facing the same issue. React native version - 0.79.0
> 
> <img alt="Image" width="452" src="https://private-user-images.githubusercontent.com/194590623/446720055-f6e39389-b66e-4fce-8389-23c0a0d390da.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDc5Nzc1NTUsIm5iZiI6MTc0Nzk3NzI1NSwicGF0aCI6Ii8xOTQ1OTA2MjMvNDQ2NzIwMDU1LWY2ZTM5Mzg5LWI2NmUtNGZjZS04Mzg5LTIzYzBhMGQzOTBkYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUyM1QwNTE0MTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YTRjYTliODY5ZDZiYzYzMTdkMWZkNTVjODYxOTJhMmQ4MDBlZDczZjg2MGFmOTBhZmZhNDc2MTIxZGY1NzVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.xwnmlVIWZ01Hem23jEw8Donhhn3qC4WGvpp4rsULZtQ">
> [@ankitpunchh](https://github.com/ankitpunchh) [@anujsachan1990](https://github.com/anujsachan1990) Did you guys found any resolution ?
> 
> In my case also, I cannot remove -fcxx-modules as adobe packages are dependent on it.
> 
> This issue should be reopened.

Hi @tw-anuj , For me it has only compiled when we removed -fcxx-modules. You also have to remove it and for the dependencies which are getting error check for the updates from them.

**Comment by vinova-vito** [comment]
I resolved the issue by adding a script in the Podfile. I haven't found a better solution for now, but you could also try running `USE_FRAMEWORKS=1 bundle exec pod install`.

```
post_install do |installer|
  require 'fileutils'

  installer.pods_project.targets.each do |t|
    next unless t.name == 'glog'
    UI.puts "🔧 Processing glog target: #{t.name}"

    root    = installer.sandbox.root
    pattern = File.join(root, 'glog', 'src', 'glog', '*.h')
    UI.puts "🔍 Searching for glog header files: #{pattern}"

    Dir.glob(pattern).each do |hfile|
      UI.puts "📄 Checking file: #{hfile}"
      FileUtils.chmod 'u+w', hfile

      text = File.read(hfile)
      new_text = if File.basename(hfile) == 'raw_logging.h'
        text
          .gsub(/^#include\s+"(?:glog\/)+log_severity\.h".*\n/, '')
          .gsub(%r{(?:glog/)+vlog_is_on\.h},    'glog/glog/vlog_is_on.h')
      else
        text
          .gsub(%r{(?:glog/)+log_severity\.h}, 'glog/glog/log_severity.h')
          .gsub(%r{(?:glog/)+vlog_is_on\.h},    'glog/glog/vlog_is_on.h')
      end

      if new_text != text
        File.write(hfile, new_text)
        UI.puts "    ✅ Updated: #{File.basename(hfile)}"
      else
        UI.puts "    ⏭️ No changes needed: #{File.basename(hfile)}"
      end
    end

    UI.puts "🔧 Finished processing glog target: #{t.name}\n\n"
  end

  react_native_post_install(
    installer,
    config[:reactNativePath],
    :mac_catalyst_enabled => false
  )
end
```

**Comment by duncannz** [comment]
The only thing that successfully resolved the issue for me was to remove the usage of glog from `react_native_assert.h`, using the following code inside the `post_install do |installer|` block:

```ruby
file_path = 'Pods/Headers/Public/React-debug/react/debug/react_native_assert.h'
lines = File.readlines(file_path)
lines_to_remove = [
  "#include <glog/logging.h>\n",
  "    LOG(ERROR) << \"react_native_assert failure: \" << #cond; \\\n",
  "    google::FlushLogFiles(google::GLOG_INFO);               \\\n",
]
lines.reject! { |line| lines_to_remove.include?(line) }
File.open(file_path, 'w') { |f| f.puts lines }
```

**Comment by ajeetfancode** [comment]
@duncannz Did you noticed any issues in logging by doing the above said changes? 
Your change successfully resolved mine, just wanted to know any other popped up issues due to this.

**Comment by quangcao-offerup** [comment]
I got same error when upgrading react native from 0.77.3 to 0.81.5 or even one minor up as 0.78.3. the project is still under old architecture and AppDelegate.mm that working just fine on 0.77.3. Tried all possible solutions above comment in this thread but not working. hoping someone has better idea to get this working. same many people in here, I could not remove  `-fcxx-modules` flag since many required dependencies need this. when comparing glog(0.3.5) when running react native 0.77 vs 0.78 or 0.81.5, they are the same under Pods folder.  
my project is run with 
- XCode 26.2
- COCOAPODS: 1.16.2

**Comment by punith-js** [comment]
@quangcao-offerup, I'm on same boat as you, please let me know if you found a fix?

**Comment by quangcao-offerup** [comment]
@punith-js so here what I get it to work by updating Podfile to update build settings and telling cocoapod to not treat glob as a module
```
installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
            # Fix glob build issue in rn 0.78+ to CocoaPods not to treat glob as a module
            if target.name == 'glog' 
                config.build_settings['DEFINES_MODULE'] = 'NO'
            end
       end
end
```

**Comment by TarasenkoArtem** [comment]
Same issue while upgrading rn from 0.77 to 0.81.5

use_modular_headers!
pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec', :modular_headers => false 

this solves the issue but brings a lot of new issue.  

<img width="385" height="953" alt="Image" src="https://github.com/user-attachments/assets/b6f8768a-df65-44fb-80ac-d5dc3e9b3a9c" />
